x/checkpointing/proposal.go

```diff
@@ -30,13 +37,19 @@ func NewProposalHandler(
 	logger log.Logger,
 	ckptKeeper CheckpointingKeeper,
 	mp mempool.Mempool,
-	txVerifier baseapp.ProposalTxVerifier,
+	bApp *baseapp.BaseApp,
+	encCfg *appparams.EncodingConfig,
 ) *ProposalHandler {
-	defaultHandler := baseapp.NewDefaultProposalHandler(mp, txVerifier)
+	defaultHandler := baseapp.NewDefaultProposalHandler(mp, bApp)
+	ckpttypes.RegisterInterfaces(encCfg.InterfaceRegistry)
+
 	return &ProposalHandler{
 		logger:                        logger,
 		ckptKeeper:                    ckptKeeper,
-		txVerifier:                    txVerifier,
+		bApp:                          bApp,
+		txEncoder:                     encCfg.TxConfig.TxEncoder(),
+		txDecoder:                     encCfg.TxConfig.TxDecoder(),
+		txBuilder:                     encCfg.TxConfig.NewTxBuilder(),
```

Comment:
doesn't we need new `encCfg.TxConfig.NewTxBuilder()` per transaction ?  i.e every time we build checkpoint we call:
```
buiilder :=  encCfg.TxConfig.NewTxBuilder()

builder.SetMsgs(checkpoint)

builder.GetTx()
```

Reply:
I think it should be fine as every time `SetMsgs` is called, the `tx.Body` will be reset https://github.com/cosmos/cosmos-sdk/blob/62212dfc04b3790f3cc052cdc2fb757b32ca97d0/x/auth/tx/builder.go#L266

Reply:
ah ok! thanks for double checking, then it is fine 👍  (it would be not fine if there would be some kind of concurrent access to prepare proposal, but I do not belive it is the case)